covenant/covenant.go

```diff
@@ -436,30 +439,64 @@ func (ce *CovenantEmulator) delegationsToBatches(dels []*types.Delegation) [][]*
 	return batches
 }
 
-func RemoveAlreadySigned(localKey *btcec.PublicKey, dels []*types.Delegation) []*types.Delegation {
-	sanitized := make([]*types.Delegation, 0, len(dels))
-	localKeyBytes := schnorr.SerializePubKey(localKey)
+// IsKeyInCommittee returns true if the covenant serialized public key is in the covenant committee of the
+// parameter in which the BTC delegation was included.
+func IsKeyInCommittee(paramCache ParamsGetter, covenantSerializedPk []byte, del *types.Delegation) bool {
+	stkParams, err := paramCache.Get(del.ParamsVersion)
+	if err != nil {
+		return false
```

Comment:
I would wrap error and return it to the caller. Then in our main covenant loop we can log that query failed.

In general, imo converting `err` to `bool` is always a bit risky as it hides from the caller error. This is especially important as this function is public.

Reply:
done at https://github.com/babylonlabs-io/covenant-emulator/pull/91/commits/1b4b6c1997ca6d8172d73804df4073a290458d17